### ISSUE TITLE
Package System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). 

### ISSUE THREAD

**Issue by Jaycecoding** [dotnet/runtime#40253]
Hi there,

I'm very new to programming and my class has started with a basic Xamarin project, however whenever I try to run the program I get this message:
Package System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). 

All that I'm trying to run is this:
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d"
             x:Class="StackLayoutTutorial.MainPage">

    <StackLayout>
        
        <Label Text="Welcome to Xamarin.Forms!" 
           HorizontalOptions="Center"
           VerticalOptions="CenterAndExpand" />
    </StackLayout>

</ContentPage>

I've tried to check/uncheck the build in the solution properties, but it doesn't seem to fix the issue. Any idea of what is causing this messsage?

**Comment by DickBaker** [comment]
Yes, "System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0" and your error likely says
Error NU1202 Package System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package System.Numerics.Vectors 4.5.0 supports:
- monoandroid10 (MonoAndroid,Version=v1.0)
- monotouch10 (MonoTouch,Version=v1.0)
- netcoreapp2.0 (.NETCoreApp,Version=v2.0)
- uap10.0.16299 (UAP,Version=v10.0.16299)
- xamarinios10 (Xamarin.iOS,Version=v1.0)
- xamarinmac20 (Xamarin.Mac,Version=v2.0)
- xamarintvos10 (Xamarin.TVOS,Version=v1.0)
- xamarinwatchos10 (Xamarin.WatchOS,Version=v1.0)

In my solution it was due to system.text.json needing dependency to System.Numerics.Vectors and resolving that from netstandard2.0 TargetFramework (that contains v4.5.0 despite being incompatible!).

Suggest you add an explicit dependency like this to your .csproj file
<PackageReference Include="System.Numerics.Vectors" Version="4.6.0-preview4.19212.13" />
so loader will resolve there instead (and compatible with  netstandard2.x).

Eventually MS will get it right !

**Comment by Jaycecoding** [comment]
> Yes, "System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0" and your error likely says
> Error NU1202 Package System.Numerics.Vectors 4.5.0 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package System.Numerics.Vectors 4.5.0 supports:
> 
> * monoandroid10 (MonoAndroid,Version=v1.0)
> * monotouch10 (MonoTouch,Version=v1.0)
> * netcoreapp2.0 (.NETCoreApp,Version=v2.0)
> * uap10.0.16299 (UAP,Version=v10.0.16299)
> * xamarinios10 (Xamarin.iOS,Version=v1.0)
> * xamarinmac20 (Xamarin.Mac,Version=v2.0)
> * xamarintvos10 (Xamarin.TVOS,Version=v1.0)
> * xamarinwatchos10 (Xamarin.WatchOS,Version=v1.0)
> 
> In my solution it was due to system.text.json needing dependency to System.Numerics.Vectors and resolving that from netstandard2.0 TargetFramework (that contains v4.5.0 despite being incompatible!).
> 
> Suggest you add an explicit dependency like this to your .csproj file (anglebrackets removed due to MD)
> PackageReference Include="System.Numerics.Vectors" Version="4.6.0-preview4.19212.13" /
> so loader will resolve there instead (and compatible with netstandard2.x).
> 
> System.Numerics.Vectors 4.5.0 dates from May 2018 (long before our current versions) so lack basic unit/integration tests.
> Eventually MS will get it right, but they should issue a health/wealth warning meantime !

Thanks for the reply Dick. Which .csproj file do I add it to? And where would the angle brackets be placed?

**Comment by DickBaker** [comment]
see att screenshot as example from an Issue and PR I was working on for
https://github.com/PacktPublishing/C-8-and-.NET-Core-3-Projects-Using-Azure-Second-Edition
![Jaycecoding](https://user-images.githubusercontent.com/2774188/91756985-9474f900-ebc5-11ea-8d72-edcbb5b4fd7d.png)

In VS2019 you can just click on the project name to display the .csproj (no need to unload and poke about)

**Comment by tannergooding** [comment]
This looks to be the same issue as https://github.com/dotnet/runtime/issues/39925#issuecomment-664622513, and a binding redirect is missing. CC. @ericstj
